client/src/components/QandA/QuestionsList/QuestionCard/Modal.jsx

```diff
@@ -20,109 +30,181 @@ const Modal = ({ setShowModal, usage, product_name, questionOrProduct_id }) => {
     }
   };
 
+  // Determines displayed text based on whether Q or A
   if (usage === 'addAnswer') {
     title = 'Submit your answer';
     subtitle = `${product_name}:`;
     title_body = 'Your Answer';
+    buttonText = 'Submit Answer';
   } else {
     title = 'Ask Your Question';
     subtitle = `About the ${product_name}:`;
     title_body = 'Your Question';
+    buttonText = 'Submit Question';
   }
 
-  const postModalInput = (endpoint, body) => {
-    // console.log('Modal Post Body: ', body);
-    axios.post(localHost + endpoint, body)
-      .catch(err => console.error('Error posting Modal PUT req: ', err));
+
+  const handleFileInputChange = (e) => {
+    const files = e.target.files;
+    const filesArray = [];
+    for (let i = 0; i < files.length; i++) {
+      filesArray.push(files[i]);
+    }
+
+    setImages(prevImages => [...filesArray, ...prevImages]);
+
+    const previewDIV = document.getElementById('QA-Modal-previews');
+    // while (previewDIV.firstChild) {
```

Comment:
mysterious while loop needed still ?

Reply:
Please keep for now. Will address by my next PR.